### ISSUE TITLE
Loading SDF file works in RDKit 2024.03.3 but not 2024.03.4

### ISSUE THREAD

**Issue by j-wags** [rdkit/rdkit#7619]
**Describe the bug**
Loading a particular SDF on MacOS worked in RDKit 2024.03.3, but crashes python in 2024.03.4

**To Reproduce**

[bace.sdf.tgz](https://github.com/user-attachments/files/16257998/bace.sdf.tgz)

```python
from rdkit import Chem

suppl = Chem.SDMolSupplier('bace.sdf')
for mol in suppl:
   print(Chem.MolToSmiles(mol))
```

Using RDKit 2024.03.3

> `CN1C(=O)[C@](c2cccc(-c3cccc(Cl)c3)c2)(C2CC2)[NH+]=C1N`

Using RDKit 2024.03.4

> `zsh: illegal hardware instruction  python`

**Expected behavior**
Either printing out a SMILES (the old behavior) or an error message. 

**Configuration (please complete the following information):**
 - RDKit version: 2024.03.4
 - OS: Mac OSX 14.4.1, 2020 M1 Macbook Air
 - Python version (if relevant): 3.11.9
 - Are you using conda? Yes
 - If you are using conda, which channel did you install the rdkit from? conda-forge

**Comment by lilyminium** [comment]
I think I'm seeing the same issue, albeit with slightly different error messages on different computers:

```
from rdkit import Chem

suppl = Chem.SDMolSupplier('CNC.sdf')
for mol in suppl:
   print(Chem.MolToSmiles(mol))
```
Personal laptop (Mac 13.5.1, 2023 Macbook Pro):
```
[1]    41700 trace trap  python
```
[CNC.sdf.txt](https://github.com/user-attachments/files/16289265/CNC.sdf.txt)

And failing CI at [nagl](https://github.com/openforcefield/openff-nagl/actions/runs/9923796791/job/27500267466) has a more similar message of:

```
Fatal Python error: Illegal instruction
```

Notably though NAGL CI is failing with build 1 of 2024.03.4  (e.g. py310h76b7c8c_1) but was [working 2 weeks ago](https://github.com/openforcefield/openff-nagl/actions/runs/9823031871/job/27120427315) with 2024.03.4, build 0 (i.e. py310h537dff2_0). I'm seeing the issue across Pythons 3.10-3.12, only on Mac.

**Comment by greglandrum** [comment]
There are no code changes on the RDKit side between the two different conda-forge builds; the difference between 2024.03.4_0 and 2023.03.4_1 is a switch in the conda-forge configuration to add numpy 2 builds.

Further info: 
- I can't reproduce this on linux
- The conda build of v2024.03.3 works fine on the Mac
- The conda build of v2024.03.4_0 works fine on the Mac
- If I build master myself on the Mac things work fine
- If I build the 2024.03 branch myself on the Mac things work fine
- the conda builds fail on the Mac with both numpy 1.26 and numpy 2.0 (that was a long shot)
-

**Comment by greglandrum** [comment]
This should be resolved in the 2024.03.5 release, which I just did today.
The conda-forge builds are already done.

**Comment by j-wags** [comment]
Wow, thanks so much for the quick investigation and action, @greglandrum. You're an inspiration for us lesser maintainers in the field. We'll try out the new packages and report back if the problem persists.
